File: Ejercicio1/Ejercicio1Python/ingresar_datos.py
from validaciones import validar_escalones
from subir_escalera import subir_escalera


def ingresar_escalones():

    while True:
        try:
            opcion_ingresada = int(input("Ingrese un numero de escalones mayor a 0: "))
            while validar_escalones(opcion_ingresada):
                opcion_ingresada = int(input("Reingrese una opción válida: "))
            break
        except ValueError:
            print("Ingresó datos inválidos")

    print("Las posibles formas de subir una escalera de", opcion_ingresada, "son", subir_escalera(opcion_ingresada),
          "formas.")


ingresar_escalones()

File: Ejercicio1/Ejercicio1Java/main/java/IngresarDatos.java
import java.util.Scanner;

public class IngresarDatos {


    public void ingresarEscalones() {

        Escalera escalera = new Escalera();
        Scanner entrada = new Scanner(System.in);

        int escalones = 0;

        while (Validaciones.validar(escalones)){
            System.out.println("Ingrese un numero de escalones mayor a 0: ");
            escalones = entrada.nextInt();
        }

        System.out.println("Una escalera de " + escalones + " escalones se puede subir de " +
                escalera.subir_escalera(escalones) + " formas.");

    }
}
